src/Symfony/Component/Lock/Store/FlockStore.php

```diff
@@ -79,12 +79,12 @@ private function lock(Key $key, $blocking)
 
         // Silence error reporting
         set_error_handler(function ($type, $msg) use (&$error) { $error = $msg; });
-        if (!$handle = fopen($fileName, 'r')) {
+        if (!$handle = fopen($fileName, 'r+')) {
```

Comment:
what about doing this, to keep working if a write permission issue is raised?
`if (!$handle = fopen($fileName, 'r+') ?: fopen($fileName, 'r')) {`
(same below of course)

Reply:
👍 that works both on my local Windows environment and on the SunOS environment :). I'll change the PR accordingly.